### ISSUE TITLE
Data.Parser takeFramedByEsc_ test failing

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/streamly#1641]
```
  Streamly/Test/Data/Parser.hs:1139:9: 
  1) Data.Parser, test for sequence parser, takeFramedByEsc_Pass
       Assertion failed (after 44 tests):
         [4401,6617,6105,3471,6877,4754,4970,5890,977,907,2716,1556,5208,2290,3304,2202,1862,5241,2548,4090,9323,2]
         stream [4401,6617,6105,3471,6877,4754,4970,5890,977,907,2716,1556,5208,2290,3304,2202,1862,5241,2548,4090,9323,1]
         list   [4401,6617,6105,3471,6877,4754,4970,5890,977,907,2716,1556,5208,2290,3304,2202,1862,5241,2548,4090,9323]
         stream length  22
         list length  21
         stream \\ list [1]
         list \\ stream []

  To rerun use: --match "/Data.Parser/test for sequence parser/takeFramedByEsc_Pass/
```

See https://github.com/composewell/streamly/runs/6569431413?check_suite_focus=true

**Comment by rnjtranjan** [comment]
It seems a special case of End element after Esc element, function is keeping the End element after Esc thats why we see 1 in the End (1 is End element 2 is Esc element).
@harendra-kumar what is the correct behavior?

**Comment by harendra-kumar** [comment]
That's the meaning of escape, anything after the escape character is interpreted literally.

**Comment by rnjtranjan** [comment]
Fix in PR #1645
